src/main/java/com/minecolonies/core/entity/ai/workers/util/Tree.java

```diff
@@ -125,6 +131,21 @@ public class Tree
      */
     private boolean netherTree = false;
 
+    /*
+     * List of tree family tags. Each tag in a given family represents a block or set of 
+     * blocks that should be considered part of the same tree.
+     */
+    private static final List<TagKey<Block>> TREE_FAMILY_TAGS = List.of(
+        ModTags.mangroveTree,
+        ModTags.oakTree,
+        ModTags.spruceTree,
+        ModTags.birchTree,
+        ModTags.darkOakTree,
+        ModTags.jungleTree,
+        ModTags.acaciaTree,
+        ModTags.cherryTree
+    );
```

Comment:
I sort of like this in theory as it preserves the ability to try to avoid treating an adjacent different-tree-type as the same tree (which can mess up sapling placement).  But it's not extensible to additional tree types added by other mods.

If you look at the `minecolonies:tree` tag there's already an entry in there that is supposed to cover all these cases -- the missing link was that the `isBlockPartOfSameTree` check defeated it.  I wonder if we should just relax that check to any #tree and then we don't need these additional tags.

Reply:
Possibly a middle ground might be to make a single new tag for the generic bee_nests tag (move it out of #trees), and then have `isBlockPartOfSameTree` check for "is it the same log block OR in that new tag" (something like #extra_tree_blocks?).

Reply:
Or actually put the new tag into #tree in its place, rather than removed entirely.  It does still need to be there for some other checks.

Reply:
two small inputs we can think about here as well:
a) I like the idea of the "extra-tree-blocks" tags that would just fine and people can datapack add anything they need
b) I kinda do like the idea of a datadriven way to identify individual trees. We could load that map from a datapack to be able to individually separate vanilla trees, modpacks can adjust if necessary (if its datapack we can add some optional tags for some modded trees) and we can have some sort of "tree_any" fallback as well?

Reply:
Yeah, that wouldn't be to solve this problem, but rather solve the problem of connecting trees together.

Reply:
also doesn't have to be part of this PR, just maybe food for thought.